From 71077d582b4c8c36a2d0db47a3de70256545cd5a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Fri, 2 Dec 2016 14:07:08 +0100
Subject: [PATCH] cogl: Do not include both GLES2 and GL headers

EGLDevice requires a define from GLES2, even when GL is used instead.
As type definitions may conflict between the two, we shouldn't include
both at the same time. Instead, provide the missing define explicitly
when not using GLES2.

https://bugzilla.gnome.org/show_bug.cgi?id=774891
---
 cogl/cogl/driver/gl/cogl-texture-2d-gl.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/cogl/cogl/driver/gl/cogl-texture-2d-gl.c b/cogl/cogl/driver/gl/cogl-texture-2d-gl.c
index d75a391..9a29c8f 100644
--- a/cogl/cogl/driver/gl/cogl-texture-2d-gl.c
+++ b/cogl/cogl/driver/gl/cogl-texture-2d-gl.c
@@ -47,8 +47,13 @@
 #include "cogl-util-gl-private.h"
 
 #if defined (COGL_HAS_EGL_SUPPORT)
-#include <GLES2/gl2.h>
-#include <GLES2/gl2ext.h>
+#include "cogl-egl-defines.h"
+#  ifndef COGL_HAS_GLES2
+/* We need this define from GLES2, but can't include the header
+   as its type definitions may conflict with the GL ones
+ */
+#  define GL_TEXTURE_EXTERNAL_OES           0x8D65
+#  endif
 #endif
 
 void
-- 
2.9.3

